pkg/core/engine/main.go

```diff
@@ -126,43 +127,65 @@ func (e *Engine) Prepare() (err error) {
 	}
 
 	err = e.LoadConfigurations()
-	if err != nil {
-		return err
+
+	if len(e.Pipelines) == 0 {
+		logrus.Errorln(err)
+		return fmt.Errorf("no valid pipeline found")
 	}
 
-	err = e.InitSCM()
+	// Don't exit if we identify at least one valid pipeline configuration
 	if err != nil {
-		return err
+		logrus.Errorln(err)
+		logrus.Infof("\n%d pipeline(s) successfully loaded\n", len(e.Pipelines))
 	}
 
+	// If one git clone fails then we exit
+	err = e.InitSCM()
+
 	return err
 }
 
 // ReadConfigurations read every strategies configuration.
 func (e *Engine) LoadConfigurations() error {
 	// Read every strategy files
+	errs := []error{}
+
 	for _, cfgFile := range GetFiles(e.Options.File) {
 
 		loadedConfiguration, err := config.New(cfgFile, e.Options.ValuesFiles, e.Options.SecretsFiles)
 
-		if err != nil && err != config.ErrConfigFileTypeNotSupported {
-			logrus.Errorf("%s\n\n", err)
+		if err == config.ErrConfigFileTypeNotSupported {
 			continue
-		} else if err == config.ErrConfigFileTypeNotSupported {
+		} else if err != nil && err != config.ErrConfigFileTypeNotSupported {
+			err = fmt.Errorf("%q - %s", cfgFile, err)
+			errs = append(errs, err)
 			continue
 		}
```

Comment:
this if/else struct starts to be hard to read. Do you mind using a switch case there?

Reply:
Sure, I definitely can

Reply:
Can't it be just to "if" without "else" here?

One "if err == config.ErrConfigFileTypeNotSupported {"
Then one "if err != nil && err != config.ErrConfigFileTypeNotSupported {"

Reply:
I would also prefer switch case as it makes things more readable

Reply:
I think it's nitpicking but I took the opportunity to a comment